predicators/nsrt_learning/strips_learning/base_strips_learner.py

```diff
@@ -43,14 +43,43 @@ def learn(self) -> List[PartialNSRTAndDatastore]:
         filtering may break it.
         """
         learned_pnads = self._learn()
+        # Remove pnads by increasing min_data_perc until harmlessness breaks.
+        if CFG.enable_harmless_op_pruning:
+            assert self._verify_harmlessness
+            assert not CFG.disable_harmlessness_check
+        # Keeps track of latest set of harmless pnads. Will interating over
+        # all min_perc_data_for_nsrts until it breaks.
+        min_harmless_pnads = learned_pnads
+        for min_perc_data_for_nsrt in range(CFG.min_perc_data_for_nsrt, 100):
```

Comment:
it would be wiser to only iterate over thresholds that make a difference, right? and since checking harmlessness is slow, this actually seems like a change worth making. can we collect up all the numbers of data across the NSRTs and then only consider thresholding at those numbers?